lib/code-manager.js

```diff
@@ -208,36 +208,40 @@ function getCurrentCodeCell(editor: atom$TextEditor) {
   let start = new Point(0, 0);
   let end = buffer.getEndPosition();
   const regexString = getRegexString(editor);
+  let cellType = "code";
 
   if (!regexString) {
-    return new Range(start, end);
+    return { range: new Range(start, end), cellType };
   }
 
   const regex = new RegExp(regexString);
   const cursor = editor.getCursorBufferPosition();
 
-  while (cursor.row < end.row && isComment(editor, cursor)) {
-    cursor.row += 1;
-    cursor.column = 0;
-  }
+  cursor.column = 0;
 
   if (cursor.row > 0) {
     buffer.backwardsScanInRange(
       regex,
       new Range(start, cursor),
-      ({ range }) => {
-        start = new Point(range.start.row + 1, 0);
+      ({ match, range }) => {
+        for (let i = 1; i < match.length; i++) {
+          if (match[i]) {
+            cellType = match[i];
```

Comment:
Great suggestion, but how should I handle aliases like md because checking for all future aliases and changing them would be cumbersome. Is there a way to map two or more strings to a certain type value?

Reply:
Nevermind I thought of a great fix. I will update this thread once I have time to implement.

Reply:
Implemented